src/main/scala/org/ergoplatform/ErgoLikeContext.scala

```diff
@@ -55,14 +58,16 @@ class ErgoLikeContext(val currentHeight: Height,
                       val initCost: Long
                  ) extends InterpreterContext {
 
-  assert(self == null || boxesToSpend.exists(box => box.id == self.id), s"Self box if defined should be among boxesToSpend")
-  assert(preHeader == null || preHeader.height == currentHeight, "Incorrect preHeader height")
-  assert(preHeader == null || java.util.Arrays.equals(minerPubkey, preHeader.minerPk.getEncoded.toArray), "Incorrect preHeader minerPubkey")
+  assert(preHeader != null)
+  assert(spendingTransaction != null)
```

Comment:
@aslesarenko I did not find anywhere in ergo that `spendingTransaction` can be null. In sigma tests, I saw null `spendingTransaction` as a shortcut to constructing `ErgoLikeContext` without the tx. Same with `preHeader`.
`assert` added just to fail early, rather than in Json encoding later.

Reply:
HF suspicious, please provide NO-HF-PROOF comment with analysis.

Reply:
Done.